### PR TITLE
Refactoring to condense disputes tests

### PR DESCRIPTION
**Pull request by BradleyOlson64**
Ran all these tests after changes to confirm they still work.

### CODE REVIEW

File: node/core/dispute-coordinator/src/tests.rs [hunk 1]
```diff
@@ -2245,68 +2167,29 @@ fn resume_dispute_without_local_statement() {
...
-					candidate_hash,
-					session,
-					true,
-				)
-				.await;
+			let supermajority_threshold =
+				polkadot_primitives::v2::supermajority_threshold(test_state.validators.len());
+			let mut statements = Vec::new();
+			// Getting votes for supermajority. Should already have two valid votes.
+			for i in (0_u32..supermajority_threshold as u32 - 2).map(|i| i + 3) {
```
Comment (tdimitrov) [inline node/core/dispute-coordinator/src/tests.rs:2174]: Maybe `for i in vec![0, 3, 4, 5, 6, 7]` is a bit more readable?

**Review by tdimitrov — COMMENTED** [review]
Good idea! Removes lots of duplicated code.

**Review by tdimitrov — APPROVED** [review]

**Review by eskimor — APPROVED** [review]
Nice!

**Comment by BradleyOlson64** [comment]
bot merge

File: node/core/dispute-coordinator/src/tests.rs [hunk 2]
```diff
@@ -2245,68 +2172,27 @@ fn resume_dispute_without_local_statement() {
...
-				.issue_explicit_statement_with_index(
-					ValidatorIndex(7),
-					candidate_hash,
-					session,
-					true,
-				)
-				.await;
+			let mut statements = Vec::new();
+			// Getting votes for supermajority. Should already have two valid votes.
+			for i in vec![3, 4, 5, 6, 7] {
```
Comment (m-cat) [inline node/core/dispute-coordinator/src/tests.rs:2177]: Late to the party here, but did you mean to leave out vote 0? (Nice PR BTW!)

Comment (BradleyOlson64) [reply node/core/dispute-coordinator/src/tests.rs:2177]: I did indeed mean to leave 0 out. It seems that 0 is our local vote. So in this test where we were supposed to trigger local participation we wouldn't need to submit a redundant explicit 0 vote.

**Pull request merged**